### PR TITLE
Web client

### PR DESCRIPTION
**Pull request by kkaris**
This PR extends the web client module to several new endpoints:
- `/api/ground`
- `/api/search`
- `/api/entity/{curie}`
- `/api/lexical`

An IPython Notebook showing simple example of how to use the different clients is added.
Tests for each client function are added.

### CODE REVIEW

File: mira/dkg/web_client.py [hunk 1]
```diff
@@ -1,104 +1,214 @@
...
     Returns
     -------
+    :
+        A list of all entities in the graph.
+    """
+    return web_client(endpoint="/lexical", method="get", api_url=api_url)
+
+
+def ground_web(
+    grounding_model: grounding.GroundRequest, api_url: Optional[str] = None
```
Comment (bgyori) [inline mira/dkg/web_client.py:154]: This definition is a little bit odd, I think we should just allow the basic grounding parameters like text, context, namespaces directly as parameters.

File: mira/dkg/web_client.py [hunk 2]
```diff
@@ -1,104 +1,214 @@
...
-    relation_direction: Literal["right", "left", "both"] = "right",
-    relation_min_hops: int = 1,
-    relation_max_hops: int = 1,
-    limit: Optional[int] = None,
-    full: bool = False,
-    distinct: bool = False,
-    api_url: str = None,
-):
-    """A wrapper that call the rest API's get_relations endpoint
+    relations_model: api.RelationQuery,
```
Comment (bgyori) [inline mira/dkg/web_client.py:79]: I just noticed the use of the api's model here as well, I guess it's more justified here given how complicated the inputs are.

File: mira/dkg/web_client.py (hunk 1 above)
Comment (kkaris) [reply mira/dkg/web_client.py]: I reverted this in https://github.com/indralab/mira/pull/38/commits/35b339c39a4f3c899711aa495f1d4381623a244e

File: mira/dkg/web_client.py (hunk 2 above)
Comment (kkaris) [reply mira/dkg/web_client.py:79]: Here I instead added a code example in the docstring so anyone using the function gets explicit instruction of how to provide the RelationQuery model. See https://github.com/indralab/mira/pull/38/commits/69dd967a107050560a6431a09a7fb8c80cc972ac

File: mira/dkg/web_client.py [hunk 3]
```diff
@@ -1,104 +1,234 @@
...
+        A list of all entities in the graph.
+    """
+    return web_client(endpoint="/lexical", method="get", api_url=api_url)
 
+
+def ground_web(
+    text: str,
+    namespaces: Optional[List[str]] = None,
+    api_url: Optional[str] = None,
+) -> Optional[grounding.GroundResults]:
```
Comment (bgyori) [inline mira/dkg/web_client.py:171]: @kkaris can we also add the context parameter here?

Comment (kkaris) [reply mira/dkg/web_client.py:171]: It's commented out: https://github.com/kkaris/mira/blob/main/mira/dkg/grounding.py#L145-L158
Not sure if we should change that. @cthoyt probably knows if we can do that or not.

**Comment by kkaris** [comment]
I need to update the notebook I realized after https://github.com/indralab/mira/pull/38/commits/35b339c39a4f3c899711aa495f1d4381623a244e, give me a couple of minutes.

File: mira/dkg/web_client.py (hunk 3 above)
Comment (cthoyt) [reply mira/dkg/web_client.py:171]: Yes, it's possible to uncomment the context parameter an enable it here without much hassle.

Comment (kkaris) [reply mira/dkg/web_client.py:171]: I'll do that

Comment (kkaris) [reply mira/dkg/web_client.py:171]: Fixed in https://github.com/indralab/mira/pull/38/commits/c61b4d685a3e8d70708425bc7bec4f2dca6f08e1

**Review by bgyori — APPROVED** [review]

**Pull request merged**